### PR TITLE
Fix TEG acting as infinite energy source on destruction

### LINKED ISSUE #29893: TEG can function as an infinite power source
## Description

It stays at the value it was generating before being destroyed.

**Reproduction**

1. Get hit by a meteor
2. Lizard round #59181

**Screenshots**

Screenshot 2024-07-10 170607
Screenshot 2024-07-10 170628

**Additional context**

### PR DESCRIPTION
**Pull request by luizwritescode**
## About the PR

Fixes #29893 
TEG will currently get stuck in the last value it generates if it any part of it gets destroyed. Read how to reproduce it and my investigation commented in the linked issue.

## Why / Balance

It's better to fix it than make the TEG indestructable

## Technical details

I go more in-depth in my comment [here](https://github.com/space-wizards/space-station-14/issues/29893#issuecomment-2226191398)

tldr:
`GeneratorUpdate` in `TegSystem` checks for being built and returns before checking for is powered, so if somehow it becomes unbuilt while it's producing power it will be stuck that way.

## Media

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

I can't think of anything that breaks by replacing the check order but I haven't tested enough to be 100% certain

**Changelog**

🆑 coffeeware
- fix: a powered TEG won't produce infinite power when destroyed

### CODE REVIEW

File: Content.Server/Power/Generation/Teg/TegSystem.cs [hunk 1]
```diff
@@ -113,6 +109,11 @@ private void GeneratorUpdate(EntityUid uid, TegGeneratorComponent component, ref
             return;
         }
 
+        var tegGroup = GetNodeGroup(uid);
+        if (tegGroup is not { IsFullyBuilt: true })
+            return;
+
+
```
Comment (ShadowCommander) [inline Content.Server/Power/Generation/Teg/TegSystem.cs:116]:
```suggestion
```

Remove extra newline.

**Review by ShadowCommander — APPROVED** [review]
Looks good. Though I'll need someone more knowledgeable to check this too.
